packages/hash/shared/src/entityStorePlugin.ts

```diff
@@ -123,6 +123,8 @@ export const entityStorePluginStateFromTransaction = (
  * session. Pass 'null' if the entity is new and the entityId is not available.
  * Do NOT change the entity's draftId mid-session - leave it as fake. If you
  * need to recall the entity's draftId, use mustGetDraftEntityForEntityId
+ *
+ * @todo should maybe always be random to handle a block being present twice
```

Comment:
A bit curious, In what scenario will an instance of block entity be present twice (same entityId). I know it's possible for a block to show up multiple times (same entityTypeId, different entityId's), but can't think of a scenario where we'd have 2 blocks with the same entityId

Reply:
It's long been an ambition in HASH to support having the same block on multiple pages, or the same block multiple times on one page (it's the "shared block" idea). It's possible we end up implementing this with multiple block entities pointing to the same data entity, but that wasn't how it was done when this was supported in HASH in the past. 

This is an out of date diff though. This has been changed now to always return a random ID.